src/componentprocessor/index.ts

```diff
@@ -4,38 +4,37 @@ import { promisify } from "util";
 import xmldoc, { XmlDocument } from "xmldoc";
 import pSettle = require("p-settle");
 const readFile = promisify(fs.readFile);
-const fg = require("fast-glob");
+import * as fg from "fast-glob";
 
 export class ComponentDefinition {
     public contents?: string;
-    public isFulfilled: boolean = true;
+    public xmlNode?: XmlDocument;
+    public name: string = "";
```

Comment:
I think `name` is being optional in this case since you don't assign any value when the promise is rejected